src/views/Editor/Toolbar/NewsValueTimeDropDown.tsx

```diff
@@ -0,0 +1,172 @@
+import { useMemo } from 'react'
+import {
+  Button, Popover,
+  PopoverTrigger,
+  PopoverContent,
+  Calendar,
+  Tabs,
+  TabsList,
+  TabsTrigger,
+  TabsContent
+} from '@ttab/elephant-ui'
+import { useRegistry } from '@/hooks'
+import { ComboBox } from '@/components/ui'
+import { dateToReadableDateTime, is12HourcycleFromLocale } from '@/lib/datetime'
+
+interface NewsValueDropDownProps {
+  duration?: string
+  end?: string
+  onChange: (duration: string | undefined, end: string | undefined) => void
+}
+export function NewsValueTimeDropDown({ duration, end, onChange }: NewsValueDropDownProps): JSX.Element {
+  const { locale, timeZone } = useRegistry()
+  const isDuration = !!duration && duration === parseInt(duration).toString(10)
+  const isEnd = !isDuration && !!end
+  const endTime = end ? new Date(end) : new Date()
+
+  let value = '∞'
+  if (isDuration) {
+    const secs = parseInt(duration)
+    const hours = Math.ceil(secs / 3600)
+    value = `${hours}h`
+  } else if (isEnd) {
+    value = 'datetime'
+  }
+
+  return (
+    <Popover>
+      <PopoverTrigger asChild>
+        <Button
+          variant="ghost"
+          className="flex h-8 p-0 px-2 mb-1 data-[state=open]:bg-muted"
+        >
+          <span className="flex items-end">{isEnd ? dateToReadableDateTime(new Date(end), locale, timeZone) : value}</span>
+        </Button>
+      </PopoverTrigger>
+
+      <PopoverContent className="w-[320px]">
```

Comment:
Since the default alignment for Popover is center, which looks best imo, the content drifts when switching between `∞`, `24h` `<datetime>`. Maybe using `align="start"` might be better?

Reply:
Perfect!